scripts/js/lib/api/Metadata.ts

```diff
@@ -10,22 +10,51 @@
 // copyright notice, and modified files need to carry a notice indicating
 // that they have been altered from the originals.
 
-export type ApiType =
+export type ApiTypeName = ApiObjectName | "module";
+export type ApiObjectName =
+  // Python API types
   | "class"
   | "method"
   | "property"
   | "attribute"
-  | "module"
   | "function"
   | "exception"
   | "data"
+  // C API types
   | "struct"
+  | "cFunction"
   | "typedef"
   | "enum"
   | "enumerator";
 
+interface ApiObjectInfo {
+  cssSelector: string;
+  tagName: "class" | "function" | "attribute";
+}
+
+/**
+ * This object defines every API type we support. The key is the name we use to
+ * refer to the API type in this script, the `cssSelector` is the CSS selector
```

Comment:
Nit that `htmlSelector` is more accurate. The selector isn't only used for CSS, e.g. HTML uses it in forms to associate labels with inputs. JS also uses it.